### ISSUE TITLE
Jetty 12.0 beta 2 HttpServletResponse::getStatus returns 0 by default

### ISSUE THREAD

**Issue by poutsma** [eclipse/jetty.project#9955]
**Jetty version(s)**
Jetty 12.0 beta 2

**Description**

Running a Servlet in version 12 beta 2, `HttpServletResponse::getStatus` returns `0` if a status has not yet been set with `setStatus`. 

According the [Section 6.1.1 of RFC 2616](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec6.1.1):

>The Status-Code element is a **3-digit integer** result code

`0` is not a 3-digit integer, and therefore should not be returned from `getStatus`.

Though I can't find a reference of a default status code in the Servlet spec, all other `HttpServletResponse` implementations—including previous versions of Jetty—have a default status code of `200`.

**How to reproduce?**

This simple Servlet will print the default status code, and prints `0` in Jetty 12, beta 2.

```java
public static class StatusCodeServlet extends HttpServlet {

	@Override
	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
		resp.setContentType("text/plain");
		resp.getWriter().write(Integer.toString(resp.getStatus()));
	}
}
```

**Comment by gregw** [comment]
Note that if left at 0, then jetty will send a 200 as 0 simple means not set and the default if not set is 200.
However, we can change the servlet APIs to always return 200, as they did in previous releases

**Comment by gregw** [comment]
Note that 0 means not set and the default when not set is 200, so 200 will be sent.
However, I will update the servlet API classes to represent 0 as 200 as they don't care about not-set

**Comment by joakime** [comment]
Merged in PR #9956
